pkg/schema/schema.go

```diff
@@ -152,6 +158,25 @@ func getType(node yamlmeta.ValueHoldingNode) (yamlmeta.Type, error) {
 	return typeOfValue, nil
 }
 
+func getExplicitDefaultValue(node yamlmeta.ValueHoldingNode) (interface{}, error) {
+	anns, err := collectAnnotations(node)
+	if err != nil {
+		return nil, NewSchemaError("Invalid schema", err)
+	}
+	annsCopy := append([]Annotation{}, anns...)
+	if len(annsCopy) == 0 {
+		return nil, nil
+	}
+
+	var defaultVal interface{}
+	for _, ann := range annsCopy {
+		if defaultAnn, ok := ann.(*DefaultAnnotation); ok {
+			defaultVal = defaultAnn.val
```

Comment:
- shouldnt be accessing private fields on objects outside of objects
- shouldnt we be deep copying this value somehow?

Reply:
Can you explain why we should consider deep copying this value?

Reply:
lets say if default is an map and that map gets modified by something else down the line (who knows if that happens in current/future code), then default value would be affected, which means next time default value is used, it will return wrong copy (aka modified value).

Reply:
Changed so that we deep copy values that satisfy the node interface.